File: Thesis/code/discord.js
const {
    ActionRowBuilder, ButtonBuilder, ButtonStyle, SlashCommandBuilder
} = require("discord.js");

module.exports = {
    data: new SlashCommandBuilder()
        .setName("pow")
        .setDescription("Umocní číslo 'num' na 'exp'.")
        .addIntegerOption(op => op.setName("num").setRequired(true))
        .addIntegerOption(op => op.setName("exp").setRequired(true)),
    async execute(interaction) {
        const num = interaction.options.getInteger("num");
        const exp = interaction.options.getInteger("exp");
        const button = new ButtonBuilder()
            .setLabel("Example")
            .setURL("https://example.com")
            .setStyle(ButtonStyle.Link);
        const row = new ActionRowBuilder().addComponents(button);
        await interaction.reply({
            content: "" + num ** exp, components: [row]
        });
    },
};

File: Thesis/code/discord.py
import discord

@discord.app_commands.command()
async def pow(self,
              itx: discord.Interaction,
              num: int,
              exp: int) -> None:
    """Umocní číslo 'num' na 'exp'."""
    view = discord.ui.View()
    view.add_item(discord.ui.Button(label="Example",
                                    url="https://example.com"))
    await itx.response.send_message(content=num ** exp,
                                    view=view)